### ISSUE TITLE
Cannot assign aliased instances

### ISSUE THREAD

**Issue by ngfreiter** [FHIR/sushi#713]
The following works fine:
```
Alias: bar = http://hl7.org/fhir/StructureDefinition/Patient

Profile: MyOb
Parent: Observation
* ^contained[0] = bar
```
but this does not work:
```
```

**Comment by ngfreiter** [comment]
Then again maybe we wouldn't expect these two cases to be the same. We have already a precedent of an alias being used to represent a string value in an assignment rule. I'm not sure how we decide whether an alias should represent a string or an inline instance.

**Comment by cmoesel** [comment]
Isn't that determination made by the type it is assigned to?  I.e., if it is a string-like type, assign as a string, otherwise treat as an instance.  We do something kind of like that w/ concepts because `foo#bar` can be a code, Coding, CodeableConcept, or Quantity units depending on what it is being assigned to.

**Comment by ngfreiter** [comment]
Yeah I think that could work. But I do think that fixing this bug may then require some significant refactoring, since currently we decide if something is an instance or not in import. If we are making that decision based on the type being assigned to, we may need to change that.

**Comment by cmoesel** [comment]
Ah.  I forgot that bit.  So, the string use...  we did that in order to support things like:
```
* compose.include[0].system = LNC
```
right?  Just trying to remember the different use cases...

**Comment by ngfreiter** [comment]
Well when an instance is assigned, its value is still just a string, but with the `isInstance` flag set. Before the ability to assign instances, aliases would always resolve to a string, and on the import side it would be impossible to tell if those strings refer to an instance or not, so the `isInstance` flag would not get set.

**Comment by AltangerelG** [comment]
I'd like to pick this up, if nobody is on it. Still reproduces on master
(`d623e96`), and the exclusion is explicit — `visitCaretValueRule` in
`FSHImporter.ts`:

```ts
caretValueRule.isInstance =
  ctx.value()?.name() != null && !this.allAliases.has(ctx.value().name().getText());
```

An alias is ruled out by construction, which is why the aliased form behaves
differently from the literal one. (Same shape in `visitFixedValueRule` and the
assignment rule above it.)

Before writing anything, the constraint I can see but can't resolve from the
code. @ngfreiter's point above — that at import time a resolved alias is just a
string, so you can't tell whether it names an Instance — is enforced downstream:
in `StructureDefinitionExporter`, `isInstance` makes the exporter fish for the
Instance and log a hard error if it finds nothing:

```ts
if (rule.isInstance) {
  const instance = instanceExporter.fishForFHIR(rule.value as string);
  if (instance == null) {
    logger.error(`Cannot assign Instance at path ${rule.path} ... Definition not found for Instance: ${rule.value}.`, ...);
```

So simply dropping `!this.allAliases.has(...)` would fix this case and make every
alias used as an ordinary string value start erroring. Three ways I can see:

1. Set `isInstance` for aliases, and have the exporter fall back to the plain
   string value instead of erroring when the fish comes up empty. Simplest, but
   it weakens a real error for the non-alias case too — a typo'd Instance name
   would go quiet.
2. Keep the error, but only for values that were not aliases, so the fallback is
   scoped to the case this issue is about.
3. Carry the ambiguity explicitly (`isInstance` as "known" / "possible") and
   resolve it at export, where fishing is available.

I lean towards 2, since it fixes the reported behaviour without loosening
diagnostics anywhere else. Do you have a preference, or is there a reason this
should stay as-is that I am not seeing?

Happy to implement whichever you'd prefer, with tests.
